lib/ruby-mpd/song.rb

```diff
@@ -24,6 +24,18 @@ def ==(another)
     self.class == another.class && self.file == another.file
   end
 
+  def to_json(*a)
+    {
+      "time" => @time,
+      "file" => @file,
+      "title" => @title,
+      "artist" => @artist,
+      "album" => @album,
+      "albumartist" => @albumartist,
+      "data" => @data
+    }.to_json(*a)
```

Comment:
```to_json``` does just work but would only return ```"\"#<MPD::Song:0x007ff066159ce0>\""```

Would a better implementation be:

```
def to_json
  Hash.new.tap do |hsh|
    instance_variables.each do |var|
      hsh[var] = self.instance_variable_get var
    end
  end.to_json
end
```

As redefining all the instance variables is not ideal.

You could then easily write some test to prove this. You are going to write some tests right?